front/app/containers/Authentication/useSteps/stepConfig/claveUnicaFlow.ts

```diff
@@ -3,16 +3,29 @@ import confirmEmail from 'api/authentication/confirm_email/confirmEmail';
 import { GetRequirements } from 'containers/Authentication/typings';
 import { askCustomFields } from './utils';
 import resendEmailConfirmationCode from 'api/authentication/confirm_email/resendEmailConfirmationCode';
+import { updateUser } from 'api/users/useUpdateUser';
 
 export const claveUnicaFlow = (
   getRequirements: GetRequirements,
   setCurrentStep: (step: Step) => void
 ) => {
   return {
     'clave-unica:email': {
-      SUBMIT_EMAIL: async (email: string) => {
-        await resendEmailConfirmationCode(email);
-        setCurrentStep('clave-unica:email-confirmation');
+      SUBMIT_EMAIL: async ({
+        email,
+        userId,
+      }: {
+        email: string;
+        userId: string;
+      }) => {
+        const { requirements } = await getRequirements();
+        if (requirements.special.confirmation === 'require') {
+          await resendEmailConfirmationCode(email);
```

Comment:
It's a bit weird to me that we're using this 'resend' endpoint in this context. This is basically submitting your email for the first time, right?

Reply:
Yes, for the first time. I agree it's weird. What is the proper way of doing it? The profile page also uses this method https://github.com/CitizenLabDotCo/citizenlab/blob/master/front/app/containers/EmailChange/UpdateEmailForm.tsx#L62

Reply:
I suppose it's the only way to send the confirmation code https://github.com/CitizenLabDotCo/citizenlab/blob/e68c703bd87a279fd485b7ad88039acb92318a28/back/app/controllers/web_api/v1/resend_codes_controller.rb#L3-L7
@jamesspeake isn't it?